### PR TITLE
[Core] Kratos-Components printing registered objs

### PR DESCRIPTION
**Pull request by philbucher**
This PR enhances the Kratos-Components, now in case a requested Object is not found, it will print all the available objs of this type
This is quite helpful for debugging (and partially done for the python-interface already, but this is the more general solution)
@tteschemacher this should help you next time :)

**The cleaning is done in a separate PR:** #3431

### CODE REVIEW

**Comment by philbucher** [comment]
the [KratosComonents specialized for VariableData don't have such a check](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/kratos_components.h#L304-L312)

Should I add it there too?

File: kratos/includes/kratos_components.h [hunk 1]
```diff
@@ -137,22 +131,23 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
     ///@{
 
     /// Turn back information as a string.
-    virtual std::string Info() const
+    static std::string Info()
```
Comment (jcotela) [inline kratos/includes/kratos_components.h]: Why are you making those static? Not that I'm complaining, but note that this breaks the `__str__` printing in python if you had it.

Comment (pooyan-dadvand) [reply kratos/includes/kratos_components.h]: Why it breaks the __str__ printing? We use the print object, isn't it?

Comment (jcotela) [reply kratos/includes/kratos_components.h]: the `__str__` mechanism is for printing class instances, so it expects a self. Once the method is static, it won't get the self. I guess that the python method can be made non-static, but then you need an instance of KratosComponents to print.

File: kratos/includes/kratos_components.h [hunk 2]
```diff
@@ -76,31 +73,28 @@ class KRATOS_API(KRATOS_CORE) KratosComponents
...
+        auto it_comp =  msComponents.find(Name);
+        if (it_comp != msComponents.end()) {
+            return *(it_comp->second);
+        }
+
+        std::stringstream err_msg;
+        err_msg << "The component \"" << Name << "\" is not registered!\n"
+                << "Maybe you need to import the application where it is defined?\n"
+                << "The following components of this type are registered:" << std::endl;
+        PrintData(err_msg);
```
Comment (philbucher) [inline kratos/includes/kratos_components.h]: @jcotela I use it here, thats why it has to be static

=> after you pointed out that this is problematic I will revert making the print-fcts static and just create an instance of `KratosComponents` here for using it

Comment (jcotela) [reply kratos/includes/kratos_components.h]: As you see, it is not a problem either way, but it is something we need to be aware of. Unfortunately, I could not find what is the equivalent of `__str__` for `print(<class>)` (as opposed to `print(<instance>)`.

Comment (pooyan-dadvand) [reply kratos/includes/kratos_components.h]: but we can export it to python as a normal method. Isn't it?

Comment (philbucher) [reply kratos/includes/kratos_components.h]: Ofc you can export it, but then the „PrintObject“ - fct I recently added cannot be used

Either way I solved this another way,  no modifications to the Print-fcts are necessary now

Comment (pooyan-dadvand) [reply kratos/includes/kratos_components.h]: How you solved it?

Comment (philbucher) [reply kratos/includes/kratos_components.h]: By creating an instance:
https://github.com/KratosMultiphysics/Kratos/blob/07c074218bd3dbd4ed2027fe6248574162b7d593/kratos/includes/kratos_components.h#L96-L97

Comment (philbucher) [reply kratos/includes/kratos_components.h]: I think this solution is fine since anyway an error is thrown

Comment (pooyan-dadvand) [reply kratos/includes/kratos_components.h]: is fine.....
Nevertheless having a private static `PrintDataImplementation` would be a cleaner....

Comment (philbucher) [reply kratos/includes/kratos_components.h]: should I change it?

If not then this is ready for merge

Comment (pooyan-dadvand) [reply kratos/includes/kratos_components.h]: It is fine as it is as we should replace it in future with registry

Comment (philbucher) [reply kratos/includes/kratos_components.h]: ok
can you approve then?

**Review by pooyan-dadvand — APPROVED** [review]
:+1:

**Pull request merged**